### PR TITLE
Editor: add sticky button in action bar

### PR DESCRIPTION
**Pull request by mtias**
image

image

Note: retaining the other sticky option in the advanced-status menu so we can compare usage.

image

### CODE REVIEW

**Comment by folletto** [comment]
Small bug on some widths:

screen shot 2016-03-21 at 12 27 10

screen shot 2016-03-21 at 12 29 43

Other than that, works well. 

I'd probably try a shorter message like: "Stick post at the top" or "Mark as top highlight" or something like that on one line. :)

File: client/post-editor/editor-sticky/index.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+				onMouseEnter={ () => this.setState( { tooltip: true } ) }
+				onMouseLeave={ () => this.setState( { tooltip: false } ) }
+				aria-label={ this.translate( 'Stick post to the front page' ) }
+				ref="stickyPostButton"
+			>
+				<Gridicon icon="bookmark" />
+				<Tooltip
+					context={ this.refs && this.refs.stickyPostButton }
+					isVisible={ this.state.tooltip }
+					position="bottom"
```
Comment (aduth) [inline client/post-editor/editor-sticky/index.jsx]: Should be `bottom left` for consistency with other action bar tooltips?

File: client/post-editor/editor-sticky/index.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+				>
+					{ this.props.post && this.props.post.sticky
+						? <span>{ this.translate( 'Marked as sticky' ) }</span>
+						: <div>
+							{ this.translate( 'Mark post as sticky' ) }
```
Comment (aduth) [inline client/post-editor/editor-sticky/index.jsx]: Could probably shorten this a bit more by removing "post", i.e. "Mark as sticky"

File: client/post-editor/editor-sticky/index.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+const EditorSticky = React.createClass( {
+	propTypes: {
+		post: React.PropTypes.object
+	},
+
+	getInitialState: function() {
+		return {
+			tooltip: false
+		}
```
Comment (aduth) [inline client/post-editor/editor-sticky/index.jsx]: Missing semi-colon.

File: client/post-editor/editor-sticky/index.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+	},
+
+	toggleStickyStatus: function() {
+		var stickyStat, stickyEventLabel;
```
Comment (aduth) [inline client/post-editor/editor-sticky/index.jsx]: Mixed `var` / `const` usage.

File: client/post-editor/editor-sticky/style.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
+.editor-sticky.button {
+	color: lighten( $gray, 20% );
+	margin-right: 12px;
+	padding: 0;
+	transition: color 200ms;
+
+	&.is-sticky{
```
Comment (aduth) [inline client/post-editor/editor-sticky/style.scss]: Space before `{`

File: client/post-editor/editor-sticky/index.jsx (hunk 1 above)
Comment (mtias) [inline client/post-editor/editor-sticky/index.jsx]: `bottom left` is looking a bit awkward, and with the space it has it should be enough.

**Comment by aduth** [comment]
I like the idea and it works well in my testing, but noting similar issues as @folletto with regards to text with, particularly for non-saved posts (where trash icon is absent).

Ideas for second line explanation text:
- "Shown first" (sub. "first" / "at top" )
- "Always top post" (sub. "top" / "first" )

File: client/post-editor/editor-sticky/index.jsx (hunk 1 above)
Comment (aduth) [inline client/post-editor/editor-sticky/index.jsx]: >and with the space it has it should be enough.

That's the problem for me though, it doesn't have much space, especially for unsaved posts.

**Comment by mtias** [comment]
Maybe just "Mark as sticky"?

**Comment by folletto** [comment]
The hard bit there is that "sticky" requires knowledge of what it really means, while "top" is clearer most of the times, but fails on themes that handle sticky post differently. So I'd probably go with a mixed wording like the above "Stick to top" so it contains both, but with a slight preference to "top" since it would still be better most of the time.

**Comment by aduth** [comment]
:+1: from me.

**Pull request merged**